File: Algorithms/1009.Complement-of-Base-10-Integer/complement_of_base_10_integer.py
class Solution:
    def bitwiseComplement(self, N: int) -> int:
        if N == 0:
            return 1
        ret = 0
        cnt = 0
        base = 1
        while N:
            cnt  +=1
            #1100110
            #0011001
            #1 0 0 
            if N % 2 == 0:
                ret += base
            base *= 2
            N //= 2
        return ret

File: Algorithms/1009.Complement-of-Base-10-Integer/complement_of_base_10_integer.go
func bitwiseComplement(N int) int {
    if N == 0 {
        return 1
    }
    res := 0
    base := 1
    for {
        if N <= 0 {
            break
        }
        if N % 2 == 0 {
            res += base
        }
        base *= 2
        N /= 2
    }
    return res
}